src/index.ts

```diff
@@ -0,0 +1,181 @@
+    actionService.create(action, handler);
+  });
+  
+  // delete all the actions matching the given `memberId`
+  fastify.delete<{ Params: IdParam }>(
+    '/members/:id/delete-actions',
+    { schema: getOne },
+    async ({ member, params: { id }, log,  }, reply) => {
+      const task = new DeleteActionsTask(member, id, actionService);
+      await runner.runSingle(task);
+
+      reply
+      .code(200)
+      .header('Content-Type', 'application/json; charset=utf-8')
+      .send( task.result )      
+    },
+  );
+
+};
+
+
+// save action (except create and delete) for the itemId with its actionType and view
+const createActionTask = async function (member: Member, itemId: string, actionType: string, view: string, runner, itemService) {
+  // get item
+  const t1 = new GetItemTask(member, itemId, itemService);
+  await runner.runSingle(t1);
+  const item = t1.result;
+  // save action
+  const extra = {memberId: member.id, itemId: item.id};
+  let action: Action = new BaseAction(member.id, item.id, member.type, item.type, actionType, view, extra);
+  const t2 = new CreateActionTask(member, action, actionService);
+  runner.runSingle(t2);
+};
+
+// save delete item action for the itemId with corresponding view
+const createDeleteActionTask = async function (member: Member, itemId: string, actionType: string, view: string, runner, itemService) {
+  // save action
+  const extra = {memberId: member.id, itemId: itemId};
+  let action: Action = new BaseAction(member.id, null, member.type, null, actionType, view, extra);
+  const t2 = new CreateActionTask(member, action, actionService);
+  runner.runSingle(t2);
+};
+
+// save action (except create and delete) for each itemId with corresponding actionType and view
+const createActionTaskMultipleItems = async function (member: Member, itemIds: [string], actionType: string, view: string, runner, itemService) {
+  // get item and save action for each item
+  for (const itemId in itemIds) {
+    const t1 = new GetItemTask(member, itemIds[itemId], itemService);
+    await runner.runSingle(t1);
+    const item = t1.result;
+    const extra = {memberId: member.id, itemId: item.id};
+    let action: Action = new BaseAction(member.id, item.id, member.type, item.type, actionType, view, extra);
+    const t2 = new CreateActionTask(member, action, actionService);
+    await runner.runSingle(t2);
+  }
+};
+
+// save delete item action for each itemId with corresponding view
+const createDeleteActionTaskMultipleItems = async function (member: Member, itemIds: [string], actionType: string, view: string, runner, itemService) {
+  // save action for each item
+  for (const itemId in itemIds) {
+    const extra = {memberId: member.id, itemId: itemIds[itemId]};
+    let action: Action = new BaseAction(member.id, null, member.type, null, actionType, view, extra);
+    const t2 = new CreateActionTask(member, action, actionService);
+    await runner.runSingle(t2);
+  }
+};
```

Comment:
It is not necessary because I have removed this and other functions. Now there is only one function to create the action/multiple actions. I have removed the specific functions of delete items actions because now they are using the `PostHook` instead of the `Fastify hook`.